primus-core/src/main/java/com/bytedance/primus/executor/ExecutorImpl.java

```diff
@@ -86,14 +86,20 @@ public ExecutorImpl(ExecutorContext executorContext, Dispatcher dispatcher) {
           .addTransition(ExecutorState.RUNNING,
               ExecutorState.RUNNING,
               ExecutorEventType.START)
+
+          // Ensure the work still starts even when being killed right after registered, since some
+          // training frameworks, such as Monolith, rely on worker with index 0 initiating the
+          // training process. Since AM retains the kill command, the worker will still be killed
+          // after being launched.
+          // TODO(hopang): simplify the state machine.
           .addTransition(ExecutorState.REGISTERED,
-              ExecutorState.EXITED_WITH_KILLED,
+              ExecutorState.STARTING,
               ExecutorEventType.KILL,
-              new KilledTransition())
+              new RegisteredToStartingTransition())
```

Comment:
Will AM receive a start event if we do nothing when receive a kill event？ When receive a kill event, we start the executor looks a little weird

Reply:
1. Yes, AM still receives executor start events. However, they are eventually ignored as the corresponding  executors have been marked as killing, and hence there is no following up chain reaction after it.

2. Agreed with the state transition here is very counter-intuitive, but it's essential as worker[0] might be killed right after registered in rare cases when there is no data task assigned to it, which could ultimately prevent the actual training from getting started. Therefore, the executor state machine has been update in this way, and hope the comment above can help mitigate some of the confusion here.

Meanwhile, the state machine can be streamlined by utilising graceful shutdown. By not aggressively killing executors under normal state transitions, there is a good chance that we can remove this weird logic entirely. Nevertheless, IMHO, the change could be too intrusive for the time being.